lib/vagrant/util/downloader.rb

```diff
@@ -29,8 +29,9 @@ class Downloader
         "vagrantup.com".freeze
       ].freeze
 
-      attr_reader :source
+      attr_accessor :source
       attr_reader :destination
+      attr_accessor :headers
```

Comment:
In the initializer we should verify that the headers are an array (`@headers = Array(@options[:headers])`)